build_runner_core/lib/src/asset/cache.dart

```diff
@@ -17,15 +20,29 @@ import 'package:glob/glob.dart';
 /// Does not implement [findAssets].
 class CachingAssetReader implements AssetReader {
   /// Cached results of [readAsBytes].
-  final _bytesContentCache = <AssetId, Future<List<int>>>{};
+  final _bytesContentCache = new LruCache<AssetId, List<int>>(
+      1024 * 1024,
+      1024 * 1024 * 512,
+      (value) => value is Uint8List ? value.lengthInBytes : value.length * 8);
+
+  /// Pending [readAsBytes] operations.
+  final _pendingBytesContentCache = <AssetId, Future<List<int>>>{};
 
   /// Cached results of [canRead].
+  ///
+  /// Don't bother using an LRU cache for this since it's just booleans.
   final _canReadCache = <AssetId, Future<bool>>{};
 
-  /// Cached results of [readAsString], per [Encoding] type used.
+  /// Cached results of [readAsString].
   ///
   /// These are computed and stored lazily using [readAsBytes].
-  final _stringContentCache = <AssetId, Map<Encoding, Future<String>>>{};
+  ///
+  /// Only files read with [utf8] encoding (the default) will ever be cached.
+  final _stringContentCache = new LruCache<AssetId, String>(
+      1024 * 1024, 1024 * 1024 * 512, (value) => value.length);
+
+  /// Pending `readAsString` operations.
+  final _pendingStringContentCache = <AssetId, Map<Encoding, Future<String>>>{};
```

Comment:
why do we still need the nested map here?

Reply:
we don't! good catch

Reply:
fixed